src/system-module.adoc

```diff
@@ -0,0 +1,257 @@
+----
+
+Produces a stream from any number of arguments, concatenating the streams produced by the nested
+expressions.
+Used to aggregate multiple values or sub-streams to pass to a single argument, or to return
+multiple results. Generally only useful with more than one subexpression.
+
+
+==== Value Constructors
+
+===== `make_string`
+
+[{nrm}]
+----
+(make_string (content text\...)) \-> string
+----
+
+Produces a non-null, unannotated string containing the concatenated content produced by the
+arguments. Nulls and annotations are discarded.
+
+TODO Probably useful to allow some other Ion scalars (at least) to allow type conversion.
+I think this would be most useful for ints, since the binary representation is more compact than
+as characters. Lobs wouldn’t work well, though.
+
+
+===== `make_symbol`
+
+[{nrm}]
+----
+(make_symbol (content text\...)) \-> symbol
+----
+
+Like `make_string` but produces a symbol.
+
+
+===== `make_list`
+
+[{nrm}]
+----
+(make_list (vals any\...)) \-> list
+----
+
+Produces a non-null, unannotated list from any number of inputs.
+Template expressions of the form `[E~1~, …, E~n~]` are equivalent to `(make_list E~1~ … E~n~)`.
+
+
+===== `make_sexp`
+
+[{nrm}]
+----
+(make_sexp (vals any\...)) \-> sexp
+----
+
+Like make_list but produces a sexp.
+
+[{nrm}]
+----
+(:make_sexp)      ⇒ ()
+(:make_sexp null) ⇒ (null)
+----
+
+
+===== `make_struct`
+
+[{nrm}]
+----
+(make_struct (kv any\...)) \-> struct
+----
+
+Produces a non-null, unannotated struct from any number of elements.
+The ``kv``s are processed in order, incrementally adding fields to an initially-empty struct.
+Various forms of ``kv``s are allowed:
+
+* A (non-null) string or symbol is treated as a field name, and MUST be followed by another value
+to comprise a key-value pair in the result. Annotations on the field name are discarded.
+* A (non-null) struct is merged into the result as-is, after discarding annotations.
+* Any other type of value evokes an expansion error.
+
+Template expressions of the form `{T~1~:E~1~, …, T~n~:E~n~}` are equivalent to
+`(make_struct (*literal* T~1~) E~1~ … (*literal* T~n~) E~n~)`, assuming that no expression E
+produces more than one value.
+In that case, the `make_struct` variant would misbehave: the second value produced by E would be
+treated as the next key.
+
+
+===== `make_decimal`
+
+[{nrm}]
+----
+(make_decimal (coefficient int) (exponent int)) \-> decimal
+----
+
+Since decimal is already compact, this is perhaps most useful in conjunction with packed arrays.
+
+TODO if the coefficient were decimal, this could re-scale values. Useful?
+
+
+===== `make_float` ❓
+
+[{nrm}]
+----
+(make_float ieee) \-> float
+----
+
+Included for completeness, but of unclear utility.
+
+TODO coerce an int or decimal to float?  Perhaps useful to use fixed-width ints to encode various float widths? This may not be useable to convert “IEEE bits” to float, since they would be converted to int before arriving here.
```

Comment:
I've always thought it would be for converting `int` to a `float`, especially small ones. (Maybe `decimal`? That one feels less clear.)

Reply:
I've cut #252 to track this (and will link from the doc)